Sources/Scheduler.swift

```diff
@@ -13,6 +13,48 @@ import Foundation
 	import CDispatch
 #endif
 
+extension Date {
+	internal func addingTimeInterval(_ interval: DispatchTimeInterval) -> Date {
+		return addingTimeInterval(interval.asTimeInterval())
+	}
+}
+
+extension DispatchTimeInterval {
+	internal func asTimeInterval() -> TimeInterval {
+		switch self {
+		case let .seconds(s):
+			return TimeInterval(s)
+		case let .milliseconds(ms):
+			return TimeInterval(TimeInterval(ms) / 1000.0)
+		case let .microseconds(us):
+			return TimeInterval( UInt64(us) * NSEC_PER_USEC ) / TimeInterval(NSEC_PER_SEC)
+		case let .nanoseconds(ns):
+			return TimeInterval(ns) / TimeInterval(NSEC_PER_SEC)
+		}
+	}
+
+	internal static prefix func -(lhs: DispatchTimeInterval) -> DispatchTimeInterval {
+		switch lhs {
+		case let .seconds(s):
+			return .seconds(-s)
+		case let .milliseconds(ms):
+			return .milliseconds(-ms)
+		case let .microseconds(us):
+			return .microseconds(-us)
+		case let .nanoseconds(ns):
+			return .nanoseconds(-ns)
+		}
+	}
+
+	/// Scales a time interval by the given scalar specified in `rhs`.
+	///
+	/// - returns: Scaled interval in microseconds
+	internal static func *(lhs: DispatchTimeInterval, rhs: Double) -> DispatchTimeInterval {
+		let seconds = lhs.asTimeInterval() * rhs
+		return .microseconds(Int(seconds * 1000 * 1000))
```

Comment:
This is falling into one of the points I wanted to fix with this approach: potential overflows.